### ISSUE TITLE
split Method Missing importFrom Statement and Folds Have Common Samples

### ISSUE THREAD

**Issue by DarioS** [SydneyBioX/BenchHub#56]
Function is in dplyr package, but it is not imported into BenchHub.
```r
> trioB1$split()
Error in if_else(stratify, "stratified", "basic") : 
  could not find function "if_else"
```
Also, the method does not work correctly for survival or numeric outcome, which is not a factor vector of classes.
```r
trioB1$split(y = rownames(trioB1$data), stratify = FALSE) 
> trioB1$splitIndices
$Fold1
 [1]  1  2  4  6  7  8  9 10 12 13 14 15 16 17 19 20 21 22 23 24 25 27 28 29 31 32 35 36 37 38 39 40 41 43 44
[36] 45 46 47 48 50 52 53 54 55 56 57 58

$Fold2
 [1]  1  2  3  5  6  9 10 11 13 14 15 16 17 18 19 20 21 22 23 24 26 28 29 30 31 32 33 34 37 38 39 40 41 42 43
[36] 44 45 46 48 49 50 51 52 53 55 57 58

$Fold3
 [1]  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 18 19 20 23 25 26 27 28 29 30 31 33 34 35 36 37 38 40 41 42
[36] 43 44 45 47 48 49 50 51 53 54 56

$Fold4
 [1]  1  2  3  4  5  7  8  9 11 12 13 15 16 17 18 19 21 22 23 24 25 26 27 28 30 32 33 34 35 36 39 41 42 43 45
[36] 46 47 48 49 51 52 54 55 56 57 58

$Fold5
 [1]  1  3  4  5  6  7  8 10 11 12 14 17 18 20 21 22 24 25 26 27 29 30 31 32 33 34 35 36 37 38 39 40 42 44 46
[36] 47 49 50 51 52 53 54 55 56 57 58
```
Samples appear in multiple folds. Needs to be the test set of each fold, not the training set.

**Comment by nick-robo** [comment]
Please keep it 2 one issue per post, it makes it very hard to track if it's not.
